### PR TITLE
provide utility to hook in connection factory

### PR DESCRIPTION
**Pull request by vangheem**

### CODE REVIEW

**Review by gitcarbs — APPROVED** [review]

File: guillotina_elasticsearch/utility.py [hunk 1]
```diff
@@ -59,10 +86,15 @@ def settings(self):
...
-            self._conn = Elasticsearch(
-                loop=self.loop, **self.settings['connection_settings'])
-        return self._conn
+        # b/w compat
+        return self.get_connection()
+
+    def get_connection(self, request=None, container=None):
+        if self._conn_util is None:
+            self._conn_util = query_utility(IConnectionFactoryUtility)
+            if self._conn_util is None:
```
Comment (bloodbare) [inline guillotina_elasticsearch/utility.py]: Why default connection factory is not defined as utility and we are using this if?

File: guillotina_elasticsearch/utility.py [hunk 2]
```diff
@@ -41,13 +43,38 @@
...
 
+class DefaultConnnectionFactoryUtility:
+    '''
+    Default uses single connection for entire application
+    '''
+
+    def __init__(self):
+        self._conn = None
+
+    def get(self, request=None, container=None, loop=None):
```
Comment (bloodbare) [inline guillotina_elasticsearch/utility.py]: I don't feel justificable the overload of sending the request everytime we need a conection. What is the use case for using also the request?

**Review by bloodbare — CHANGES_REQUESTED** [review]

File: guillotina_elasticsearch/utility.py (hunk 1 above)
Comment (vangheem) [reply guillotina_elasticsearch/utility.py]: so it can be overridden

File: guillotina_elasticsearch/utility.py (hunk 2 above)
Comment (vangheem) [reply guillotina_elasticsearch/utility.py]: It's not required and this implementation does not use it.

File: guillotina_elasticsearch/utility.py (hunk 1 above)
Comment (bloodbare) [reply guillotina_elasticsearch/utility.py]: I mean DefaultConnectionFactory should be defined as default utility so query_utility gets it?

Comment (vangheem) [reply guillotina_elasticsearch/utility.py]: Yes, if we do that then you can get configuration conflict errors

Comment (vangheem) [reply guillotina_elasticsearch/utility.py]: one sec, I'll do that. It shouldn't happen. Something else is going on.

**Review by ableeb — APPROVED** [review]

**Pull request merged**